docs/current/guides/757394-use-service-containers.md

```diff
@@ -54,167 +54,290 @@ Service containers come with the following built-in features:
+```
 
 </TabItem>
-<TabItem value="Node.js">
+<TabItem value="Python">
 
-- Use the `withExposedPort()` method to set ports that the service container will listen on. Dagger checks the health of each exposed port prior to running any clients that use the service, so that clients don't have to implement their own polling logic.
-- Use the `endpoint()` method to create a string address to a service container's port. You can either specify a port or let Dagger pick the first exposed port.
+```python file=./snippets/use-services/bind-services-2/main.py
+```
 
-Here's an example:
+</TabItem>
+</Tabs>
 
-<Embed id="cx-3lzMDn5i" />
+### Bind host services to containers
+
+Starting with Dagger v0.8.8, you can bind host services to client containers.
+
+This type of service binding is useful when you need services on the host to communicate with one or more containers. One use case is for testing, where you need to be able to spin up ephemeral containers to run tests.
+
+Here is an example of how to instantiate host services with Dagger.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+TODO: The Dagger pipeline calls `Host.tunnel(service).start`. By default, Dagger lets the operating system randomly choose which port to use based on the available ports on the host's side. Finally, a call to `endpoint` gets the final `addr` with whichever port is bound.
+
+</TabItem>
+<TabItem value="Node.js">
+
+TODO: The Dagger pipeline calls `Host.tunnel(service).start`. By default, Dagger lets the operating system randomly choose which port to use based on the available ports on the host's side. Finally, a call to `endpoint` gets the final `addr` with whichever port is bound.
 
 </TabItem>
 <TabItem value="Python">
 
-- Use the `with_exposed_port()` method to set ports that the service container will listen on. Dagger checks the health of each exposed port prior to running any clients that use the service, so that clients don't have to implement their own polling logic.
-- Use the `endpoint()` method to create a string address to a service container's port. You can either specify a port or let Dagger pick the first exposed port.
+TODO: The Dagger pipeline calls `Host.tunnel(service).start`. By default, Dagger lets the operating system randomly choose which port to use based on the available ports on the host's side. Finally, a call to `endpoint` gets the final `addr` with whichever port is bound.
+
+</TabItem>
+</Tabs>
+
+### Bind container services to the host
+
+Starting with Dagger v0.8.8, you can bind client containers to the host.
+
+This type of container networking is useful when you need a client container to access services running on the host.
 
 Here's an example:
 
-<Embed id="OPUGXdIujRC" />
+<Tabs groupId="language">
+<TabItem value="Go">
+
+TODO: This example calls `Host.service([]PortForward)` to create a service that proxies traffic through the host to the configured ports. It then sets the service binding on the client container to the host.
```

Comment:
@vito can you please provide an example? In any one language is fine, I will convert for the other SDKs.